fvm/evm/emulator/emulator.go

```diff
@@ -165,6 +167,32 @@ func (bl *BlockView) RunTransaction(
 	return res, proc.commitAndFinalize()
 }
 
+// DryRunTransaction run unsigned transaction without persisting the state
+func (bl *BlockView) DryRunTransaction(
+	tx *gethTypes.Transaction,
+	from gethCommon.Address,
+) (*types.Result, error) {
+	proc, err := bl.newProcedure()
+	if err != nil {
+		return nil, err
+	}
+
+	// here we ignore the error because the only reason an error occurs
+	// is if the signature is invalid, which we don't care about since
+	// we use the from address as the signer
+	msg, _ := gethCore.TransactionToMessage(
+		tx,
```

Comment:
I think we should handle the error here.

Reply:
I can handle it but the only error returned will be related to an invalid signature which we should ignore because it will always be an invalid signature, so what specifically do you suggest I do?

Reply:
I think it might be good to make sure that the error is an invalid signature before we ignore it..

Reply:
the problem is there are multiple un-typed errors returned based on the transaction type, so not sure how I would assert that

Reply:
I didn't find the error type at first, I see now there's a signature error type, will update